test/models/bot/alegre_4_test.rb

```diff
@@ -0,0 +1,86 @@
+require_relative '../../test_helper'
+
+class Bot::Alegre4Test < ActiveSupport::TestCase
+  def setup
+    super
+    ft = DynamicAnnotation::FieldType.where(field_type: 'language').last || create_field_type(field_type: 'language', label: 'Language')
+    at = create_annotation_type annotation_type: 'language', label: 'Language'
+    @field = create_field_instance annotation_type_object: at, name: 'language', label: 'Language', field_type_object: ft, optional: false
+    @bot = create_alegre_bot(name: "alegre", login: "alegre")
+    @bot.approve!
+    p = create_project
+    p.team.set_languages = ['en','pt','es']
+    p.team.save!
+    @bot.install_to!(p.team)
+    @team = p.team
+  end
+
+  def teardown
+    super
+  end
+
+  test "should not send bad text for langid" do
+    text = "platform~1234"
+    pm1 = create_project_media team: @team, quote: text
+    Bot::Alegre.stubs(:get_language_from_alegre).returns("und")
+    assert_equal Bot::Alegre.get_language_from_text(pm1, text), "und"
+    Bot::Alegre.unstub(:get_language_from_alegre)
+  end
+
+  test "should not send blank text for langid" do
+    text = ""
+    pm1 = create_project_media team: @team, quote: text
+    Bot::Alegre.stubs(:get_language_from_alegre).returns("en")
+    assert_equal Bot::Alegre.get_language_from_text(pm1, text), "und"
+    Bot::Alegre.unstub(:get_language_from_alegre)
+  end
+
+  test "should send text for langid" do
+    text = "anything else"
+    pm1 = create_project_media team: @team, quote: text
+    Bot::Alegre.stubs(:get_language_from_alegre).returns("en")
+    assert_equal Bot::Alegre.get_language_from_text(pm1, text), "en"
+    Bot::Alegre.unstub(:get_language_from_alegre)
+  end
+
+  test "/text/similarity/search/ request should not be called for bad title" do
+    text = "instagram-cekfakta-2023-25562004"
+    pm1 = create_project_media team: @team, quote: text
+    Bot::Alegre.stubs(:request).raises("Request method called when it should not be")
+    assert_nothing_raised do
+        Bot::Alegre.get_items_from_similar_text(@team, text)
+    end
+    Bot::Alegre.unstub.stubs(:request)
+  end
+
+  test "/text/similarity/search/ request should not be called for blank title" do
+    text = ""
+    pm1 = create_project_media team: @team, quote: text
+    Bot::Alegre.stubs(:request).raises("Request method called when it should not be")
+    assert_nothing_raised do
+        Bot::Alegre.get_items_from_similar_text(@team, text)
+    end
+    Bot::Alegre.unstub.stubs(:request)
+  end
+
```

Comment:
Please add a test with a valid text value that should make it through the if statement and call  `request`. We can use `assert_raises` (https://www.rubydoc.info/gems/minitest/Minitest%2FAssertions:assert_raises) and the same pattern as the other tests. This test should go with the other tests for `get_items_from_similar_text` at https://github.com/meedan/check-api/blob/801d44b98a33d60e61e29f3da91512ccf11efaff/test/models/bot/alegre_2_test.rb#L525-L527